src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceData/CounterSet.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Microsoft.Win32;
+
+namespace System.Diagnostics.PerformanceData
+{
+    /// <summary>
+    /// CounterSet is equivalent to "Counter Object" in native performance counter terminology,
+    /// or "Counter Category" in previous framework releases. It defines a abstract grouping of
+    /// counters, where each counter defines measurable matrix. In the new performance counter
+    /// infrastructure, CounterSet is defined by GUID called CounterSetGuid, and is hosted inside
+    /// provider application, which is also defined by another GUID called ProviderGuid.
+    /// </summary>    
+    public class CounterSet : IDisposable
+    {
+        private static readonly bool s_platformNotSupported = (Environment.OSVersion.Version.Major < 6);
```

Comment:
Is this even needed for .NET Core? .NET Core 2.1 is [supported on Windows 7 SP1+](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1-supported-os.md), which has a major version of 6. I'd get rid of it and the associated resource string about requiring Windows Vista and above.

Reply:
fixed